### PR TITLE
Introduce code style and reformat existing code #403

### LINKED ISSUE #403: Introduce a code-formatter plugin for repository code
**Is your feature request related to a problem? Please describe.**

https://github.com/google/google-java-format

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by kobylynskyi**
Closes #403.

@jxnu-liguobin please take a look

### CODE REVIEW

**Comment by jxnu-liguobin** [comment]
> Closes #403.
> 
> @jxnu-liguobin please take a look

Only added XML, no any plugin. Will it be automatically formatted when building?

**Comment by kobylynskyi** [comment]
> Only added XML, no any plugin. Will it be automatically formatted when building?

We have many independent sub-projects for individual plugins and example projects and each one would require the plugin. So I think it is much easier to add it to at the IDE level.

**Comment by jxnu-liguobin** [comment]
> > Only added XML, no any plugin. Will it be automatically formatted when building?
> 
> We have many independent sub-projects (for individual plugins and example projects) and each one of them would require the plugin. So I think it is much easier to add code style at the IDE level.

Our original intention is to solve the problem that the development style of multiple people is not uniform, which leads to the time consumption of increasing the review code. Can idea level support be detected in CI? I am not clear about this. If not, it is impossible to detect whether PR follows the format.

**Comment by kobylynskyi** [comment]
It is hard to understand form PR of code conforms to the code style. But it is possible

**Comment by jxnu-liguobin** [comment]
> It is hard to understand form PR of code conforms to the code style. But it is possible

There are plug-ins to do this, which are very common in scala.  
Ci can be easily implemented with plug-in https://github.com/zio/zio-redis/pull/303/checks?check_run_id=2044482587

**Comment by kobylynskyi** [comment]
@jxnu-liguobin I've enabled CheckStyle verification in GitHub CI. COnfig is residing in the `config/checkstyle` folder.

File: plugins/sbt/graphql-java-codegen-sbt-plugin/project/build.properties [hunk 1]
```diff
@@ -1 +1 @@
-sbt.version = 1.4.7
\ No newline at end of file
+sbt.version=1.4.7
```
Comment (jxnu-liguobin) [inline plugins/sbt/graphql-java-codegen-sbt-plugin/project/build.properties:1]: All Scala files should be ignored, and sbt example does not need to be formatted.

File: src/main/java/com/kobylynskyi/graphql/codegen/kotlin/KotlinGraphQLTypeMapper.java [hunk 2]
```diff
@@ -128,36 +164,13 @@ public ValueMapper getValueMapper() {
         return valueMapper;
     }
 
-    /**
-     * Thi method was used in kotlin template, do not remove it.
-     *
-     * @param kotlinType type get from Type template
-     * @return default value
-     */
-    public static String defaultValueKotlinPrimitive(String kotlinType) {
```
Comment (jxnu-liguobin) [inline src/main/java/com/kobylynskyi/graphql/codegen/kotlin/KotlinGraphQLTypeMapper.java:137]: This unused method must not be removed by formatting.

File: plugins/sbt/graphql-java-codegen-sbt-plugin/project/build.properties (hunk 1 above)
Comment (kobylynskyi) [reply plugins/sbt/graphql-java-codegen-sbt-plugin/project/build.properties:1]: Ok. I will do it tomorrow

File: gradlew [hunk 3]
```diff
@@ -26,156 +26,156 @@
...
 # Escape application args
-save () {
-    for i do printf %s\\n "$i" | sed "s/'/'\\\\''/g;1s/^/'/;\$s/\$/' \\\\/" ; done
-    echo " "
+save() {
+  for i; do printf %s\\n "$i" | sed "s/'/'\\\\''/g;1s/^/'/;\$s/\$/' \\\\/"; done
+  echo " "
 }
-APP_ARGS=`save "$@"`
+APP_ARGS=$(save "$@")
```
[the remark this answers is not in the document]
Comment (kobylynskyi) [reply gradlew:178]: Yeah, makes sense

[the remark this answers is not in the document]
Comment (kobylynskyi) [reply gradlew]: Reverted

File: plugins/sbt/graphql-java-codegen-sbt-plugin/project/build.properties (hunk 1 above)
Comment (kobylynskyi) [reply plugins/sbt/graphql-java-codegen-sbt-plugin/project/build.properties:1]: Actually, what's wrong with this particular change? Checkstyle did not touch scala files. Just `.properties` files

Comment (jxnu-liguobin) [reply plugins/sbt/graphql-java-codegen-sbt-plugin/project/build.properties:1]: Does not conform to the scala project common format. But it's no problem.

**Review by jxnu-liguobin — APPROVED** [review]

**Pull request merged**